app/governance/components/ProposalTable/ProposalTable.tsx

```diff
@@ -0,0 +1,136 @@
+"use client";
+import React, { useEffect, useState } from 'react';
+import Filter from '../Filter/Filter';
+import styles from './ProposalTable.module.scss';
+import { Proposal } from '@/hooks/gov/interfaces/proposal';
+import { formatDate, formatProposalStatus, formatProposalType } from '@/utils/gov/formatData';
+import Text from '@/components/text';
+import Button from '@/components/button/button';
+import Link from 'next/link';
+import { useRouter } from 'next/navigation';
+
+
+interface TableProps {
+  proposals: Proposal[];
+}
+
+const ProposalTable: React.FC<TableProps> = ({ proposals }) => {
+
+  
+  const router = useRouter();
+  const [currentFilter, setCurrentFilter] = useState('All');
+  const [filteredProposals, setFilteredProposals] = React.useState<Proposal[]>(proposals);
```

Comment:
dont need this as a state at all. Just filter the proposals from the filter itself. Unnecessary set state twice with this